optuna/samplers/_lazy_random_state.py

```diff
@@ -0,0 +1,29 @@
+from typing import Optional
+
+import numpy
+
+
+class LazyRandomState:
+
+    """Lazy Random Sate class.
+
+
+    This is a class to initialize just before using random state
```

Comment:
The starting point of this PR is this issue, but I think this class resolves more general problem. How about to write more general style, i. e. “initialize just before using random state to prevent same random state when deepcopy is applied to the instance of sampler."